pkg/trace/info/stats.go

```diff
@@ -288,6 +284,51 @@ func (s *SpansMalformed) String() string {
 	return mapToString(s.tagValues())
 }
 
+// maxAbsPriority specifies the absolute maximum priority for stats purposes. For example, with a value
+// of 10, the range of priorities reported will be [-10, 10].
+const maxAbsPriority = 10
+
+// samplingPriorityStats holds the sampling priority metrics that will be reported every 10s by the agent.
+type samplingPriorityStats struct {
+	// counts holds counters for each priority in position maxAbsPriorityValue + priority.
+	// Priority values are expected to be in the range [-10, 10].
+	counts [maxAbsPriority*2 + 1]int64
+}
+
+// CountSamplingPriority increments the counter of observed traces with the given sampling priority by 1
+func (s *samplingPriorityStats) CountSamplingPriority(p sampler.SamplingPriority) {
+	if p > (-1*maxAbsPriority) && p < maxAbsPriority {
+		atomic.AddInt64(&s.counts[maxAbsPriority+p], 1)
+	}
+}
+
+// reset sets stats to 0
+func (s *samplingPriorityStats) reset() {
+	for i := 0; i < len(s.counts); i++ {
+		atomic.StoreInt64(&s.counts[i], 0)
+	}
+}
+
+// update absorbs recent stats on top of existing ones.
+func (s *samplingPriorityStats) update(recent *samplingPriorityStats) {
+	for i := 0; i < len(s.counts); i++ {
+		atomic.AddInt64(&s.counts[i], atomic.LoadInt64(&recent.counts[i]))
+	}
+}
+
+// TagValues returns a map with the number of traces that have been observed for each priority tag
+func (s *samplingPriorityStats) TagValues() map[string]int64 {
+	statsPerTag := make(map[string]int64)
+	for i := 0; i < len(s.counts); i++ {
+		count := atomic.LoadInt64(&s.counts[i])
+		priorityTag := fmt.Sprint(i - maxAbsPriority)
+		if count > 0 {
+			statsPerTag[priorityTag] = count
+		}
```

Comment:
```suggestion
		if count > 0 {
			statsPerTag[strconv.Itoa(i - maxAbsPriority)] = count
		}
```
Or, maybe you prefer:
```go
		if count > 0 {
            tag := strconv.Itoa(i - maxAbsPriority)
			statsPerTag[tag] = count
		}
```
`strconv.Itoa` (https://pkg.go.dev/strconv#Itoa) will be much faster. It's not an issue here, but generally better practice. `fmt.Sprint` is a world-famous offender so best avoid as much as possible 🙂 

Personally I'd also recommend reducing the map's name to simply `stats`. It is very obvious that this identifier is the star and return value of the method, which is well documented, so its role is not ambiguous at all. I don't think it needs to be fully spelled out.

Reply:
Thank you! will apply these changes